odoo/_monkeypatches/__init__.py

```diff
@@ -11,29 +14,69 @@ def set_timezone_utc():
         time.tzset()
 
 
-def patch_all():
+class PatchImportHook:
+    """Register hooks that are run on import."""
+
+    def __init__(self):
+        self.hooks = {}
+
+    def add_hook(self, fullname: str, hook):
+        """Register a hook after a module is loaded.
+        If already loaded, run hook immediately."""
+        self.hooks[fullname] = hook
+        if fullname in sys.modules:
+            hook()
+
+    def find_spec(self, fullname, path=None, target=None):
+        if (hook := self.hooks.get(fullname)) is None:
+            return
```

Comment:
is my comment right? if so can we keep it? I don't remember how the import system works and this comment might help me the next time I've to read this file
```suggestion
            return  # let python uses another import hook to import this fullname
```